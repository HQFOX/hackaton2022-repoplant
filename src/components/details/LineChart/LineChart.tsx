import React, { useEffect } from "react";
import { useTranslation } from "react-i18next";
import { HvLinechart } from "@hv/uikit-react-core";

const LineChart = ({ data, getLineData }) => {
  const { t } = useTranslation();

  useEffect(() => {
    getLineData();
  }, [getLineData]);

  return data.length ? (
    <HvLinechart
      title={t("components.details.chart.title")}
      subtitle={t("components.details.chart.subtitle")}
      data={data}
      xAxisTitle={t("components.details.chart.xAxisTitle")}
      yAxisTitle={t("components.details.chart.yAxisTitle")}
    />
  ) : null;
};

export default LineChart;
