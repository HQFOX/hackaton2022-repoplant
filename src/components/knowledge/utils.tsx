import {
  AssetInventoryMetadata,
  AssetInventoryConfiguration,
} from "@hitachivantara/uikit-react-core";

const metadata: AssetInventoryMetadata[] = [
  {
    id: "id1",
    accessor: "title",
    cellType: "alpha-numeric",
  },
  {
    id: "id2",
    accessor: "project",
    cellType: "alpha-numeric",
  },
];

const viewConfiguration: AssetInventoryConfiguration = {
  breakpoints: { xs: 12, sm: 6, md: 4, lg: 3, xl: 3 },
  columnConfiguration: [
    {
      style: { paddingLeft: "10px", textAlign: "start" },
    },
    {
      title: "Title",
      style: { textAlign: "start", minWidth: "120px" },
    },
    {
      title: "Project",
      style: { textAlign: "start", minWidth: "80px" },
    },
    {
      title: "Information Type",
      style: { textAlign: "start", minWidth: "60px" },
    },
    {
      title: "Uploaded By",
      style: { textAlign: "start", minWidth: "60px" },
    },
    {
      title: "Date",
      style: {
        textAlign: "start",
      },
    },
  ],
};

const configuration = {
  metadata,
  viewConfiguration,
};

const actions = [{ id: "view", label: "View", disabled: false }];

export { configuration, actions };
