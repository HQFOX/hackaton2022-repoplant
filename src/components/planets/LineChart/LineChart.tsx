import React, { useEffect } from "react";
import { useTranslation } from "react-i18next";
import { HvLinechart } from "@hv/uikit-react-core";

const LineChart = ({ data, getLineData }) => {
  const { t } = useTranslation();

  useEffect(() => {
    getLineData();
  }, [getLineData]);

  return data.length ? (
    <HvLinechart
      title={t("components.planets.chart.title")}
      subtitle={t("components.planets.chart.subtitle")}
      data={data}
      xAxisTitle={t("components.planets.chart.xAxisTitle")}
      yAxisTitle={t("components.planets.chart.yAxisTitle")}
    />
  ) : null;
};

export default LineChart;
