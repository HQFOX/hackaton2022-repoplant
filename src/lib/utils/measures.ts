export const avg = (values: number[]): number => {
  if (!values || values.length === 0) {
    return 0;
  }
  return (
    values.reduce((a, b) => {
      return a + b;
    }) / values.length
  );
};
