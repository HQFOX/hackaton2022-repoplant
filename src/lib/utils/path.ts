import { Page } from "typings/pages";

const getSelection = (data: Page[], pathname: string): Page | undefined => {
  let selection;

  data.forEach((item) => {
    if (item.path === pathname) selection = { ...item };
    else if (item.data && pathname.indexOf(item.path) > -1)
      selection = getSelection(item.data, pathname);
  });

  return selection;
};

export { getSelection };
