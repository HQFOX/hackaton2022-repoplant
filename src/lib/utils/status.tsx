import React, { lazy } from "react";
import { Level5 } from "@hitachivantara/uikit-react-icons";
import { Semantic } from "@hitachivantara/uikit-react-lab";

export const getStatus = (status: any) => {
  const Level = `Level${status}`;
  const statusObj: {
    Icon: React.ReactNode;
    sema: Semantic;
  } = { Icon: <Level5 />, sema: "sema5" };
  return statusObj;
};
