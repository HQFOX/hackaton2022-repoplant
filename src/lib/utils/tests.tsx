import React from "react";
import { MemoryRouter } from "react-router-dom";
import GlobalProvider from "lib/providers/GlobalProvider";

export const withProvider = <P extends {}>(
  Component: React.ComponentType<P>
): React.FC<P> => (props) => (
  <MemoryRouter initialEntries={[{ pathname: "/", key: "testKey" }]}>
    <GlobalProvider>
      <Component {...props} />
    </GlobalProvider>
  </MemoryRouter>
);
