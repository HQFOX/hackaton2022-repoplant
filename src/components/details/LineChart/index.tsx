import { bindActionCreators, Dispatch } from "redux";
import { connect } from "react-redux";
import { getLineData } from "store/data/thunks";
import { withStyles } from "@material-ui/core";
import { AppState } from "typings/state";
import styles from "./styles";
import LineChart from "./LineChart";

const mapStateToProps = ({ data }: AppState) => ({
  data: data.line,
});

const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      getLineData,
    },
    dispatch
  );

export default withStyles(styles, { name: "LineChart" })(
  connect(mapStateToProps, mapDispatchToProps)(LineChart)
);
