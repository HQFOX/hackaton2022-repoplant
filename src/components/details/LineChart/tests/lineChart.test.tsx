import React from "react";
import { render } from '@testing-library/react';
import { withStoreProvider } from "lib/utils/tests";
import LineChart from "..";

const data = [{ x: ["Q1"], y: [100], name: "Sales Target" }];

describe("<LineChart />", () => {
  let component;

  const WithProvider = withStoreProvider(() => <LineChart />, {
    data: { line: data }
  });

  beforeEach(() => {
    component = render(<WithProvider />);
  });

  it("should be defined", () => {
    expect(component).toBeDefined();
  });

  it("matches the snapshot", () => {
    expect(component.container).toMatchSnapshot();
  });

  /*it("should be rendered", () => {
    const lineChart = component.find(LineChart);
    expect(lineChart.length).toBe(1);
  });*/
});
