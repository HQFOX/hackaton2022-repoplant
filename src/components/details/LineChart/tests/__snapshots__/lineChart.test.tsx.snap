// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineChart /> matches the snapshot 1`] = `
<div>
  <div
    class="HvChart-root"
  >
    <div>
      <div
        class="js-plotly-plot"
        style="position: relative;"
      >
        <div
          class="plot-container plotly"
        >
          <div
            class="user-select-none svg-container"
            style="position: relative; width: 700px; height: 450px;"
          >
            <svg
              class="main-svg"
              height="450"
              style="background: transparent;"
              width="700"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            >
              <defs
                id="defs-35a8e5"
              >
                <g
                  class="clips"
                >
                  <clippath
                    class="plotclip"
                    id="clip35a8e5xyplot"
                  >
                    <rect
                      height="350"
                      width="570"
                    />
                  </clippath>
                  <clippath
                    class="axesclip"
                    id="clip35a8e5x"
                  >
                    <rect
                      height="450"
                      width="570"
                      x="50"
                      y="0"
                    />
                  </clippath>
                  <clippath
                    class="axesclip"
                    id="clip35a8e5y"
                  >
                    <rect
                      height="350"
                      width="700"
                      x="0"
                      y="50"
                    />
                  </clippath>
                  <clippath
                    class="axesclip"
                    id="clip35a8e5xy"
                  >
                    <rect
                      height="350"
                      width="570"
                      x="50"
                      y="50"
                    />
                  </clippath>
                </g>
                <g
                  class="gradients"
                />
              </defs>
              <g
                class="bglayer"
              >
                <rect
                  class="bg"
                  height="350"
                  style="fill: rgb(0, 0, 0); fill-opacity: 0; stroke-width: 0;"
                  width="570"
                  x="50"
                  y="50"
                />
              </g>
              <g
                class="draglayer cursor-crosshair"
              >
                <g
                  class="xy"
                >
                  <rect
                    class="nsewdrag drag cursor-pointer"
                    data-subplot="xy"
                    height="350"
                    style="fill: transparent; stroke-width: 0; pointer-events: all;"
                    width="570"
                    x="50"
                    y="50"
                  />
                  <rect
                    class="nwdrag drag cursor-pointer"
                    data-subplot="xy"
                    height="20"
                    style="fill: transparent; stroke-width: 0; pointer-events: none;"
                    width="20"
                    x="30"
                    y="30"
                  />
                  <rect
                    class="nedrag drag cursor-pointer"
                    data-subplot="xy"
                    height="20"
                    style="fill: transparent; stroke-width: 0; pointer-events: none;"
                    width="20"
                    x="620"
                    y="30"
                  />
                  <rect
                    class="swdrag drag cursor-pointer"
                    data-subplot="xy"
                    height="20"
                    style="fill: transparent; stroke-width: 0; pointer-events: none;"
                    width="20"
                    x="30"
                    y="400"
                  />
                  <rect
                    class="sedrag drag cursor-pointer"
                    data-subplot="xy"
                    height="20"
                    style="fill: transparent; stroke-width: 0; pointer-events: none;"
                    width="20"
                    x="620"
                    y="400"
                  />
                  <rect
                    class="ewdrag drag cursor-pointer"
                    data-subplot="xy"
                    height="20"
                    style="fill: transparent; stroke-width: 0; pointer-events: none;"
                    width="456"
                    x="107"
                    y="400.5"
                  />
                  <rect
                    class="wdrag drag cursor-pointer"
                    data-subplot="xy"
                    height="20"
                    style="fill: transparent; stroke-width: 0; pointer-events: none;"
                    width="57"
                    x="50"
                    y="400.5"
                  />
                  <rect
                    class="edrag drag cursor-pointer"
                    data-subplot="xy"
                    height="20"
                    style="fill: transparent; stroke-width: 0; pointer-events: none;"
                    width="57"
                    x="563"
                    y="400.5"
                  />
                  <rect
                    class="nsdrag drag cursor-pointer"
                    data-subplot="xy"
                    height="280"
                    style="fill: transparent; stroke-width: 0; pointer-events: none;"
                    width="20"
                    x="29.5"
                    y="85"
                  />
                  <rect
                    class="sdrag drag cursor-pointer"
                    data-subplot="xy"
                    height="35"
                    style="fill: transparent; stroke-width: 0; pointer-events: none;"
                    width="20"
                    x="29.5"
                    y="365"
                  />
                  <rect
                    class="ndrag drag cursor-pointer"
                    data-subplot="xy"
                    height="35"
                    style="fill: transparent; stroke-width: 0; pointer-events: none;"
                    width="20"
                    x="29.5"
                    y="50"
                  />
                </g>
              </g>
              <g
                class="layer-below"
              >
                <g
                  class="imagelayer"
                />
                <g
                  class="shapelayer"
                />
              </g>
              <g
                class="cartesianlayer"
              >
                <g
                  class="subplot xy"
                >
                  <g
                    class="layer-subplot"
                  >
                    <g
                      class="shapelayer"
                    />
                    <g
                      class="imagelayer"
                    />
                  </g>
                  <g
                    class="gridlayer"
                  >
                    <g
                      class="x"
                    >
                      <path
                        class="xgrid crisp"
                        d="M0,50v350"
                        style="stroke: rgb(204, 206, 208); stroke-opacity: 1; stroke-width: 1px;"
                        transform="translate(335,0)"
                      />
                    </g>
                    <g
                      class="y"
                    >
                      <path
                        class="ygrid crisp"
                        d="M50,0h570"
                        style="stroke: rgb(204, 206, 208); stroke-opacity: 1; stroke-width: 1px;"
                        transform="translate(0,312.5)"
                      />
                      <path
                        class="ygrid crisp"
                        d="M50,0h570"
                        style="stroke: rgb(204, 206, 208); stroke-opacity: 1; stroke-width: 1px;"
                        transform="translate(0,225)"
                      />
                      <path
                        class="ygrid crisp"
                        d="M50,0h570"
                        style="stroke: rgb(204, 206, 208); stroke-opacity: 1; stroke-width: 1px;"
                        transform="translate(0,137.5)"
                      />
                    </g>
                  </g>
                  <g
                    class="zerolinelayer"
                  />
                  <path
                    class="xlines-below"
                  />
                  <path
                    class="ylines-below"
                  />
                  <g
                    class="overlines-below"
                  />
                  <g
                    class="xaxislayer-below"
                  />
                  <g
                    class="yaxislayer-below"
                  />
                  <g
                    class="overaxes-below"
                  />
                  <g
                    class="plot"
                    clip-path="url('#clip35a8e5xyplot')"
                    transform="translate(50, 50)"
                  >
                    <g
                      class="scatterlayer mlayer"
                    >
                      <g
                        class="trace scatter trace09d116"
                        style="stroke-miterlimit: 2; opacity: 1;"
                      >
                        <g
                          class="fills"
                        />
                        <g
                          class="errorbars"
                        />
                        <g
                          class="lines"
                        />
                        <g
                          class="points"
                        />
                        <g
                          class="text"
                        />
                      </g>
                    </g>
                  </g>
                  <g
                    class="overplot"
                  />
                  <path
                    class="xlines-above crisp"
                    d="M50,400.5H620"
                    style="fill: none; stroke-width: 1px; stroke: rgb(204, 206, 208); stroke-opacity: 1;"
                  />
                  <path
                    class="ylines-above crisp"
                    d="M0,0"
                    style="fill: none;"
                  />
                  <g
                    class="overlines-above"
                  />
                  <g
                    class="xaxislayer-above"
                  >
                    <g
                      class="xtick"
                    >
                      <text
                        data-math="N"
                        data-unformatted="Q1"
                        style="font-family: \\"Open Sans\\", verdana, arial, sans-serif; font-size: 10px; fill: rgb(65, 65, 65); fill-opacity: 1; white-space: pre;"
                        text-anchor="middle"
                        transform="translate(335,0)"
                        x="0"
                        y="413"
                      >
                        Q1
                      </text>
                    </g>
                  </g>
                  <g
                    class="yaxislayer-above"
                  >
                    <g
                      class="ytick"
                    >
                      <text
                        data-math="N"
                        data-unformatted="99"
                        style="font-family: \\"Open Sans\\", verdana, arial, sans-serif; font-size: 10px; fill: rgb(65, 65, 65); fill-opacity: 1; white-space: pre;"
                        text-anchor="end"
                        transform="translate(0,400)"
                        x="49"
                        y="3.5"
                      >
                        99
                      </text>
                    </g>
                    <g
                      class="ytick"
                    >
                      <text
                        data-math="N"
                        data-unformatted="99.5"
                        style="font-family: \\"Open Sans\\", verdana, arial, sans-serif; font-size: 10px; fill: rgb(65, 65, 65); fill-opacity: 1; white-space: pre;"
                        text-anchor="end"
                        transform="translate(0,312.5)"
                        x="49"
                        y="3.5"
                      >
                        99.5
                      </text>
                    </g>
                    <g
                      class="ytick"
                    >
                      <text
                        data-math="N"
                        data-unformatted="100"
                        style="font-family: \\"Open Sans\\", verdana, arial, sans-serif; font-size: 10px; fill: rgb(65, 65, 65); fill-opacity: 1; white-space: pre;"
                        text-anchor="end"
                        transform="translate(0,225)"
                        x="49"
                        y="3.5"
                      >
                        100
                      </text>
                    </g>
                    <g
                      class="ytick"
                    >
                      <text
                        data-math="N"
                        data-unformatted="100.5"
                        style="font-family: \\"Open Sans\\", verdana, arial, sans-serif; font-size: 10px; fill: rgb(65, 65, 65); fill-opacity: 1; white-space: pre;"
                        text-anchor="end"
                        transform="translate(0,137.5)"
                        x="49"
                        y="3.5"
                      >
                        100.5
                      </text>
                    </g>
                    <g
                      class="ytick"
                    >
                      <text
                        data-math="N"
                        data-unformatted="101"
                        style="font-family: \\"Open Sans\\", verdana, arial, sans-serif; font-size: 10px; fill: rgb(65, 65, 65); fill-opacity: 1; white-space: pre;"
                        text-anchor="end"
                        transform="translate(0,50)"
                        x="49"
                        y="3.5"
                      >
                        101
                      </text>
                    </g>
                  </g>
                  <g
                    class="overaxes-above"
                  />
                </g>
              </g>
              <g
                class="polarlayer"
              />
              <g
                class="ternarylayer"
              />
              <g
                class="geolayer"
              />
              <g
                class="funnelarealayer"
              />
              <g
                class="pielayer"
              />
              <g
                class="treemaplayer"
              />
              <g
                class="sunburstlayer"
              />
              <g
                class="glimages"
              />
            </svg>
            <div
              class="gl-container"
            />
            <svg
              class="main-svg"
              height="450"
              width="700"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            >
              <defs
                id="topdefs-35a8e5"
              >
                <g
                  class="clips"
                />
              </defs>
              <g
                class="indicatorlayer"
              />
              <g
                class="layer-above"
              >
                <g
                  class="imagelayer"
                />
                <g
                  class="shapelayer"
                />
              </g>
              <g
                class="infolayer"
              >
                <g
                  class="g-gtitle"
                />
                <g
                  class="g-xtitle"
                >
                  <text
                    class="xtitle"
                    data-math="N"
                    data-unformatted="components.details.chart.xAxisTitle"
                    style="font-family: \\"Open Sans\\", verdana, arial, sans-serif; font-size: 10px; fill: rgb(65, 65, 65); opacity: 1; font-weight: normal; white-space: pre;"
                    text-anchor="middle"
                    x="335"
                    y="440"
                  >
                    components.details.chart.xAxisTitle
                  </text>
                </g>
                <g
                  class="g-ytitle"
                >
                  <text
                    class="ytitle"
                    data-math="N"
                    data-unformatted="components.details.chart.yAxisTitle"
                    style="font-family: \\"Open Sans\\", verdana, arial, sans-serif; font-size: 10px; fill: rgb(65, 65, 65); opacity: 1; font-weight: normal; white-space: pre;"
                    text-anchor="middle"
                    transform="rotate(-90,20,225)"
                    x="20"
                    y="225"
                  >
                    components.details.chart.yAxisTitle
                  </text>
                </g>
              </g>
              <g
                class="menulayer"
              />
              <g
                class="zoomlayer"
              />
            </svg>
            <div
              class="modebar-container"
              style="position: absolute; top: 0px; right: 0px; width: 100%;"
            />
            <svg
              class="main-svg"
              height="450"
              width="700"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            >
              <g
                class="hoverlayer"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
